src/assets/wise5/components/draw/draw-student/draw-student.component.ts

```diff
@@ -0,0 +1,437 @@
+    const image = this.generateImageFromCanvas();
+    this.NotebookService.addNote(image, null, [componentStateId]);
+  }
+
+  generateImageFromCanvas(): any {
+    const canvasBase64Image = this.getCanvas().toDataURL('image/png');
+    return this.UtilService.getImageObjectFromBase64String(canvasBase64Image);
+  }
+
+  getCanvas(): any {
+    const canvas: any = $(`#${this.drawingToolId} canvas`);
+    if (canvas != null && canvas.length > 0) {
+      return canvas[0];
+    } else {
+      return null;
+    }
+  }
+
+  /**
+   * Create a component state with the merged student responses
+   * @param componentStates an array of component states
+   * @return a component state with the merged student responses
+   */
+  createMergedComponentState(componentStates: any[]): any {
+    const allDrawCanvasObjects = [];
+    for (const componentState of componentStates) {
+      switch (componentState.componentType) {
+        case 'Draw':
+          allDrawCanvasObjects.push(...this.getDrawCanvasObjects(componentState));
+          const backgroundImage = this.getDrawCanvasBackground(componentState);
+          if (backgroundImage != null) {
+            this.setBackgroundImage(backgroundImage);
+          }
+          break;
+        case 'ConceptMap':
+        case 'Embedded':
+        case 'Graph':
+        case 'Label':
+        case 'Table':
+          this.importWorkAsBackground(componentState);
+      }
+    }
+    return this.createComponentStateWithCanvasObjects(allDrawCanvasObjects);
+  }
+
+  createComponentStateWithCanvasObjects(canvasObjects: any[]): any {
+    const drawData: any = JSON.parse(this.getDrawData());
+    drawData.canvas.objects = canvasObjects;
+    const componentState: any = this.NodeService.createNewComponentState();
+    componentState.studentData = {
+      drawData: JSON.stringify(drawData)
+    };
+    return componentState;
+  }
+
+  getDrawCanvasObjects(componentState: any): any[] {
+    const studentData = componentState.studentData;
+    const drawDataJSON = JSON.parse(studentData.drawData);
+    return drawDataJSON.canvas.objects;
+  }
+
+  getDrawCanvasBackground(componentState: any): string {
+    const studentData = componentState.studentData;
+    const drawDataJSON = JSON.parse(studentData.drawData);
+    if (drawDataJSON.canvas.backgroundImage != null) {
+      return drawDataJSON.canvas.backgroundImage.src;
+    }
+    return null;
+  }
+
+  importWorkAsBackground(componentState: any): void {
+    const connectedComponent = this.UtilService.getConnectedComponentByComponentState(
+      this.componentContent,
+      componentState
+    );
+    if (connectedComponent.importWorkAsBackground) {
+      this.setComponentStateAsBackgroundImage(componentState);
+    }
+  }
+
+  setComponentStateAsBackgroundImage(componentState: any): void {
+    this.generateImageFromComponentState(componentState).then((image: any) => {
+      this.setBackgroundImage(image.url);
+    });
+  }
```

Comment:
These two functions (importWorkAsBackground(), setComponentStateAsBackgroundImage()) and two functions in lable-student.component.ts (mergeBackgroundGeneratingComponentState() and setComponentStateAsBackgroundImage()) are similar. Extract to parent component-student.component.ts?